### ISSUE TITLE
Warnings with conditions other than boolean cannot be disabled

### ISSUE THREAD

**Issue by RobinCodeX** [TypeScriptToLua/TypeScriptToLua#1341]
> - Added a warning when trying to use a type in a condition that can never be false in Lua, such as numbers or strings. (Only when strictNullChecks is enabled.)

I refer to the update log to set `strictNullChecks` to false, but it still triggers this warning.

https://github.com/TypeScriptToLua/TypeScriptToLua/blob/e1a1fa699742e95f7480d8fe98579a48df6c424e/src/transformation/utils/typescript/types.ts#L103

🤔Is it possible that `||` should be changed to `&&`？

my tsconfig:
```json
{
    "compilerOptions": {
        "rootDir": ".",
        "target": "esnext",
        "lib": ["esnext"],
        "moduleResolution": "node",
        "types": [
            "lua-types/jit",
            "@moddota/dota-lua-types",
            "typescript-to-lua/language-extensions"
        ],
        "strict": true,
        "strictNullChecks": false,
        "experimentalDecorators": true,
        "plugins": [
            {
                "transform": "@moddota/dota-lua-types/transformer"
            }
        ]
    },
    "include": ["**/*.ts"],
    "tstl": {
        "luaTarget": "JIT"
    }
}
```
